roles/ovirt-datacenter-cleanup/tasks/main.yml

```diff
@@ -0,0 +1,128 @@
+# Cleanup VMPools
+- name: Find existing VMPools
+  ovirt_vmpools_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing VMPools
+  debug:
+    var: ovirt_vm_pools
+
+- name: Cleanup VMPools
+  ovirt_vms:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_vm_pools }}"
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup VMs
+- name: Find existing VMs
+  ovirt_vms_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing VMs
+  debug:
+    var: ovirt_vms
+
+- name: Cleanup VMs
+  ovirt_vms:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_vms }}"
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup Templates
+- name: Find existing Templates
+  ovirt_templates_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing Templates
+  debug:
+    var: ovirt_templates
+
+- name: Cleanup Templates
+  ovirt_templates:
+    state: absent
+    auth: "{{ ovirt_auth }}"
+    name: "{{ ovirt_item.name }}"
+  with_items: "{{ ovirt_templates }}"
+  when: ovirt_item.name != 'Blank'
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup Disks
+- name: Find existing Disks
+  ovirt_disks_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
+
+- name: Show existing Disks
+  debug:
+    var: ovirt_disks
+
+- name: Cleanup Disks
+  ovirt_disks:
+    state: absent
+    name: "{{ ovirt_item.name }}"
+    auth: "{{ ovirt_auth }}"
+  with_items: "{{ ovirt_disks }}"
+  loop_control:
+    loop_var: ovirt_item
+
+# Cleanup Storages
+- name: Find existing Storage Domains
+  ovirt_storage_domains_facts:
+    auth: "{{ ovirt_auth }}"
+    pattern: "datacenter={{ data_center_name }}"
+    fetch_nested: true
```

Comment:
Do you really need to always use `fetch_nested: true` ?

Reply:
I actually don't need it from what I can see. I 'll verify and repost with all comments adjusted.